File: Python/Input.py
def numput(s, type):
    while True:
        try:
            return type(input(s))
        except:
            print("Invalid input! Try again.")

def main():
    print(input("Enter your name: "))
    print(numput("Enter your age: ", int))
    print(numput("Enter your height: ", float))

if __name__ == "__main__":
    main()

File: Cpp/Input.cpp
#include <iostream>
#include <string>
#include <sstream>
#include <type_traits>

static std::string input(const char *s)
{
    std::cout << s;
    std::string buffer;
    std::getline(std::cin, buffer);
    return buffer;
}

template <typename T>
static T input(const char *s)
{
    while (true) 
    {
        std::istringstream ss (input(s));
        T value {};
        if (!(ss >> value) || (ss >> std::ws, !ss.eof())) 
        {
            std::cout << "Invalid input! Try again.\n";
            continue;
        }
        return value;
    }
}

int main()
{
    std::cout << input("Enter your name: ") << '\n';
    std::cout << input<int>("Enter your age: ") << '\n';
    std::cout << input<float>("Enter your height: ") << '\n';
    return 0;    
}